### PR TITLE
Backend models + tests

### PR DESCRIPTION
**Pull request by hachiyuki8**
Implemented all four models under the exact same specification as listed [here](https://github.com/CMU-17-356/cmu-17-356-final-project-s23-flurry/wiki/Object-Data-Model)
Also set up linter and CI for tests

Notes:
- Checks for the uniqueness of ids and validity of references (foreign keys) will be done in controller tests

### CODE REVIEW

File: server/tests/models/record.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { describe, it } from 'mocha';
+
+describe('Testing Record model', function() {
+  it('1. Creating new record', function(done) {
+    const r = new Record({id: 'r1', latitude: -87.46, longitude: 180, timestamp: new Date('2022-02-22T10:08:00.000-05:00'), driver_id: 'd1', slip_score: 50.88});
+    r.validate().then(() => {
+      expect(r.id).to.equal('r1');
+      expect(r.latitude).to.equal(-87.46);
+      expect(r.longitude).to.equal(180);
+      expect(r.timestamp.valueOf()).to.equal(new Date(1645542480000).valueOf());
```
Comment (meghrajeev) [inline server/tests/models/record.js:14]: The test for timestamp is a bit fragile and could break if the timezone of the test environment changes. Consider using a library like moment.js to parse and compare timestamps to avoid potential issues with timezones.

File: server/tests/models/record.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      expect(r.driver_id).to.equal('d1');
```
Comment (meghrajeev) [inline server/tests/models/record.js:15]: Can we add a test for checking if  driver_id fields are missing or empty. The driver ids are required for analytics.

File: server/tests/models/record.js (hunk 1 above)
Comment (hachiyuki8) [reply server/tests/models/record.js:14]: There shouldn't be any issue with time zones, because neither the epoch timestamp nor the earlier line initialized from date with time zone offset specified depends on the actual time zone of the device the test is running on.
